### PR TITLE
Listado de empleados para administradores

### PR DESCRIPTION
**Pull request by CamiloRamirezR**
## Release Notes

- **New Features**
	- Introduced invitation status and date fields for employee management.
	- Added a new endpoint for retrieving a paginated list of employees with role-based access control.
  
- **Bug Fixes**
	- Enhanced error handling for employee listing and pagination.

- **Tests**
	- Implemented comprehensive tests for the employee listing API and updated tests for Firestore employee handling, including new cases for invitation status management.

### CODE REVIEW

File: repositories/firestore/employee.py [hunk 1]
```diff
@@ -88,15 +108,59 @@ def create(self, employee: Employee) -> None:
...
+        Returns:
+        tuple[list[Employee], int]: Lista de empleados y el número total de empleados.
+
+        """
+        client_ref = self.db.collection('clients').document(client_id)
+        query: Query = client_ref.collection('employees').order_by('invitation_date', direction=Query.DESCENDING)
+
+        # Saltar los elementos para llegar a la página solicitada
+        offset = (page_number - 1) * page_size
+        query = query.offset(offset).limit(page_size)
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply repositories/firestore/employee.py]: Wouldn't this refactor add too much additional complexity?

File: blueprints/employee.py [hunk 2]
```diff
@@ -90,3 +95,43 @@ def post(self, employee_repo: EmployeeRepository = Provide[Container.employee_re
...
         return json_response(employee_to_dict(employee), 201)
+
+
+@class_route(blp, '/api/v1/employees')
+class EmployeeList(MethodView):
+    init_every_request = False
+
+    @requires_token
+    def get(self, token: dict[str, Any], employee_repo: EmployeeRepository = Provide[Container.employee_repo]) -> Response:
+        # Verificar si el usuario tiene permisos de administrador
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply blueprints/employee.py]: I pushed a fix in commit 261d8dd, please review it.

File: blueprints/employee.py [hunk 3]
```diff
@@ -90,3 +95,43 @@ def post(self, employee_repo: EmployeeRepository = Provide[Container.employee_re
...
+        if token['role'] != Role.ADMIN.value:
+            return error_response('Forbidden: You do not have access to this resource.', 403)
+
+        client_id = token['cid']
+
+        # Parámetros de paginación opcionales
+        page_size = request.args.get('page_size', default=5, type=int)
+        page_number = request.args.get('page_number', default=1, type=int)
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply blueprints/employee.py:114]: I pushed a fix in commit 261d8dd, please review it.

File: blueprints/employee.py [hunk 4]
```diff
@@ -90,3 +95,43 @@ def post(self, employee_repo: EmployeeRepository = Provide[Container.employee_re
...
+
+        # Validar el valor de page_size
+        if page_size not in [5, 10, 20]:
+            return error_response('Invalid pageSize. Allowed values are 5, 10, 20.', 400)
+
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply blueprints/employee.py]: I pushed a fix in commit 261d8dd, please review it.

File: tests/repositories/firestore/test_employee.py [hunk 5]
```diff
@@ -149,13 +163,18 @@ def test_create(self, *, assigned: bool) -> None:
         del client_dict['id']
         self.client.collection('clients').document(client.id).set(client_dict)
 
+        # Establecemos una fecha de invitación inicial
+        invitation_date = datetime.now(UTC)
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply tests/repositories/firestore/test_employee.py:167]: I pushed a fix in commit 261d8dd, please review it.

File: tests/repositories/firestore/test_employee.py [hunk 6]
```diff
@@ -112,10 +120,16 @@ def test_get_found(self, *, assigned: bool) -> None:
             email=self.faker.unique.email(),
             password=pbkdf2_sha256.hash(self.faker.password()),
             role=self.faker.random_element(list(Role)),
+            invitation_status=InvitationStatus.UNINVITED,  # Establecemos un estado de invitación inicial
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply tests/repositories/firestore/test_employee.py]: I pushed a fix in commit 261d8dd, please review it.

File: tests/repositories/firestore/test_employee.py [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import os
 from dataclasses import asdict
+from datetime import UTC, datetime
```
[the remark this answers is not in the document]
Comment (CamiloRamirezR) [reply tests/repositories/firestore/test_employee.py:3]: I pushed a fix in commit 261d8dd, please review it.

**Review by VivianaOj — APPROVED** [review]

**Pull request merged**